jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/Jps.java

```diff
@@ -0,0 +1,136 @@
+/*[INCLUDE-IF Sidecar18-SE]*/
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package openj9.tools.attach.diagnostics;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import com.sun.tools.attach.AttachNotSupportedException;
+import com.sun.tools.attach.spi.AttachProvider;
+import com.sun.tools.attach.VirtualMachine;
+import com.sun.tools.attach.VirtualMachineDescriptor;
+
+/**
+ * Java Process Status
+ * A tool for listing Java processes and their information.
+ *
+ */
+public class Jps {
+
+	private static final String SUN_JAVA_COMMAND = "sun.java.command"; //$NON-NLS-1$
+	private static final String SUN_JVM_ARGS = "sun.jvm.args"; //$NON-NLS-1$
+	private static boolean printApplicationArguments;
+	private static boolean printJvmArguments;
+	private static boolean noPackageName;
+	private static boolean vmidOnly;
+	/**
+	 * Print a list of Java processes and information about them.
+	 * @param args Arguments to the application
+	 */
+	public static void main(String[] args) {
+		int rc = 0;
+		parseArguments(args);
+		AttachProvider theProvider = AttachProvider.providers().get(0);
+		if (null == theProvider) {
```

Comment:
If there are no providers, I expect the line above threw `IndexOutOfBoundsException`.

Reply:
This has not been fixed.